internal/providers/terraform/parser.go

```diff
@@ -754,3 +765,64 @@ func parseKnownModuleRefs(resData map[string]*schema.ResourceData, conf gjson.Re
 		}
 	}
 }
+
+type projectParserOption func(metadata *singleProjectParser)
+
+func addProviderTypeMetadata(p schema.Provider) projectParserOption {
+	return func(s *singleProjectParser) {
+		s.metadata.Type = p.Type()
+		p.AddMetadata(s.metadata)
+	}
+}
+
+func useSnapshot(t bool) projectParserOption {
+	return func(s *singleProjectParser) {
+		s.usesSnapshot = t
+		s.parser.usePriorState(t)
+	}
+}
+
+type singleProjectParser struct {
+	ctx          *config.ProjectContext
+	usesSnapshot bool
+	parser       *Parser
+	metadata     *schema.ProjectMetadata
+}
+
+func newSingleProjectParser(path string, ctx *config.ProjectContext, parserOptions ...projectParserOption) *singleProjectParser {
+	p := &singleProjectParser{
+		ctx:      ctx,
+		parser:   NewParser(ctx, false),
+		metadata: config.DetectProjectMetadata(path),
+	}
+
+	for _, f := range parserOptions {
+		f(p)
+	}
+
+	return p
+}
+
+func (s *singleProjectParser) newProject() *schema.Project {
+	name := schema.GenerateProjectName(s.metadata, s.ctx.RunContext.Config.EnableDashboard)
+
+	return schema.NewProject(name, s.metadata)
+}
+
+func (s *singleProjectParser) parseJSON(j []byte, usage map[string]*schema.UsageData) (*schema.Project, error) {
+	project := s.newProject()
+
+	pastResources, resources, err := s.parser.parseJSON(j, usage)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse resources from provided JSON structure %w", err)
+	}
+
+	project.Resources = resources
+
+	project.HasDiff = !s.ctx.ProjectConfig.TerraformUseState || s.usesSnapshot
+	if project.HasDiff {
+		project.PastResources = pastResources
```

Comment:
Just curious why do we set `HasDiff` if `usesSnapshot` is true?  I think `usesSnpshot` means pastResources is always nil.

Reply:
this has been removed as was referencing old logic, where we loaded the snapshot resources into `PastResources`